src/Swarm/Language/Parse.hs

```diff
@@ -286,36 +297,49 @@ mkStmt (Just x) = Binder x
 --     App (App (TDef a) (TDef b)) (TDef x)
 --   This function fix that by converting the Apps into Binds, so that it results in:
 --     Bind a (Bind b (Bind c))
-fixDefMissingSemis :: Term -> Term
+fixDefMissingSemis :: Syntax -> Syntax
 fixDefMissingSemis term =
   case nestedDefs term [] of
     [] -> term
-    defs -> foldr1 (TBind Nothing) defs
+    -- TODO: figure out what should be the Syntax Location of this rewrite
```

Comment:
I guess there is nothing to do here because we shouldn't need location of the bind term (since we already have the one for t1 and t2)

Reply:
Alright, thank you for the suggestion. I've added a Monoid Location instance.